pyccel/codegen/printing/pycode.py

```diff
@@ -660,15 +666,22 @@ def _print_For(self, expr):
     def _print_FunctionalFor(self, expr):
         body, iterators = self._find_functional_expr_and_iterables(expr)
         lhs = self._print(expr.lhs)
-        body = self._print(body.rhs)
-        for_loops = ' '.join(['for {} in {}'.format(self._print(idx), self._print(iters))
-                        for idx, iters in zip(expr.indices, iterators)])
+        condition = ''
+        if isinstance(body, Assign):
+            body = self._print(body.rhs)
+        else:
+            assert isinstance(body, ListAppend)
+            body = self._print(body.args[0])
+
+        for_loops = ' '.join([f'for {self._print(idx)} in {self._print(iters)}{" if " + self._print(condition.blocks[0].condition) if condition else ""}'
+                             for idx, iters, condition in zip(expr.indices, iterators, expr.conditions)])
 
-        name = self._aliases.get(type(expr),'array')
-        if name == 'array':
+        name = expr.target_type
+        if 'array' in str(name):
```

Comment:
Why:
`if 'array' in str(name):`
not:
`if 'array' == str(name):`
?

What additional case are you catching here?

Reply:
Fixed! I need to check if it is a list or a numpy array